code/game/turfs/unsimulated/beach.dm

```diff
@@ -34,8 +34,31 @@
 /turf/unsimulated/beach/water
 	name = "Shallow Water"
 	icon_state = "seashallow"
+	var/obj/machinery/poolcontroller/linkedcontroller = null
 	water_overlay_image = "water_shallow"
 
+/turf/unsimulated/beach/water/Entered(atom/movable/AM, atom/OldLoc)
+	. = ..()
+	if(!linkedcontroller)
+		return 
+	if(ishuman(AM))

```

Comment:
I think you want to be checking `ismob` on all of these, because temperature still affects non-humans